### PR TITLE
Fix e2e network test flakes

### LINKED ISSUE #9041: flake: error reading /etc/cni/net.d/something
Starting to see a lot of this one:
```
Running: podman [options] network rm -f testNetThreeCNI2
Error: in /etc/cni/net.d/testNetSingleCNI.conflist: error reading /etc/cni/net.d/testNetSingleCNI.conflist: open /etc/cni/net.d/testNetSingleCNI.conflist: no such file or directory
```
...which then cascades into:
```
Running: podman [options] network create testNetThreeCNI2
Error: the network name testNetThreeCNI2 is already used
```
ISTR this being a race condition between `network create` and `network rm`, and thought there was an open issue about it already, but I can't find it now.

## Podman network [It] podman inspect container two CNI networks (container not running)
* fedora : int podman fedora-32 root host
  * PR #8030
    * [10-15 06:47](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5326005927673856/html/int-podman-fedora-32-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---1)
    * [10-15 06:47](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5326005927673856/html/int-podman-fedora-32-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---2)
    * [10-15 06:47](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5326005927673856/html/int-podman-fedora-32-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---3)
* gce_instance:fedora : int podman fedora-33 root container
  * PR #8956
    * [01-14 12:47](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5083534094237696/html/int-podman-fedora-33-root-container.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---1)
    * [01-14 12:47](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5083534094237696/html/int-podman-fedora-33-root-container.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---2)
    * [01-14 12:47](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5083534094237696/html/int-podman-fedora-33-root-container.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---3)
* gce_instance:fedora : int podman fedora-33 root host
  * PR #8486
    * [11-26 06:41](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6182710563569664/html/int-podman-fedora-33-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---1)
    * [11-26 06:41](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6182710563569664/html/int-podman-fedora-33-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---2)
    * [11-26 06:41](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6182710563569664/html/int-podman-fedora-33-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---3)
* gce_instance:prior-fedora : int podman fedora-32 root container
  * PR #9014
    * [01-20 11:29](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6745074539167744/html/int-podman-fedora-32-root-container.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---1)
    * [01-20 11:29](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6745074539167744/html/int-podman-fedora-32-root-container.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---2)
    * [01-20 11:29](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6745074539167744/html/int-podman-fedora-32-root-container.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---3)
* gce_instance:prior-fedora : int podman fedora-32 root host
  * PR #8306
    * [11-11 11:43](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5064390275760128/html/int-podman-fedora-32-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---1)
    * [11-11 11:43](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5064390275760128/html/int-podman-fedora-32-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---2)
    * [11-11 11:43](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5064390275760128/html/int-podman-fedora-32-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---3)
* gce_instance:ubuntu : int podman ubuntu-20 root host
  * PR #8493
    * [11-27 06:39](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5928499217170432/html/int-podman-ubuntu-20-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---1)
    * [11-27 06:39](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5928499217170432/html/int-podman-ubuntu-20-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---2)
    * [11-27 06:39](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5928499217170432/html/int-podman-ubuntu-20-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---3)
* gce_instance:ubuntu : int podman ubuntu-2010 root host
  * PR #8794
    * [12-21 08:03](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5747335080706048/html/int-podman-ubuntu-2010-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---1)
    * [12-21 08:03](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5747335080706048/html/int-podman-ubuntu-2010-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---2)
    * [12-21 08:03](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5747335080706048/html/int-podman-ubuntu-2010-root-host.log.html#t--podman-inspect-container-two-CNI-networks--container-not-running---3)
```

### PR DESCRIPTION
**Pull request by Luap99**
1. Use random network names in the e2e tests
Unlike the container storage all e2e test are using the same cni
config directory. This causes problems if the network name already
exists. Using random names will make the second run pass even if the
first failed. This is only done to prevent full CI failures.

2. Move the cni lock file into the cni config dir
Commit(fe3faa5) introduced a lock file for network create/rm calls.
There is a problem with the location of the lock file. The lock file was
stored in the tmpdir. Running multiple podman network create/remove
commands in parallel with different tmpdirs made the lockfile inaccessible
to the other process, and so parallel read/write operations to the cni
config directory continued to occur. This scenario happened frequently
during the e2e tests and caused some flakes.

Fixes #9041

### CODE REVIEW

File: test/e2e/network_test.go [hunk 1]
```diff
@@ -450,6 +450,7 @@ var _ = Describe("Podman network", func() {
 		net := "macvlan" + stringid.GenerateNonCryptoID()
 		nc := podmanTest.Podman([]string{"network", "create", "--macvlan", "lo", net})
 		nc.WaitWithDefaultTimeout()
+		defer podmanTest.removeCNINetwork(net)
```
Comment (edsantiago) [inline test/e2e/network_test.go:453]: Good catch

**Review by edsantiago — COMMENTED** [review]
Nice work, thank you!

**Comment by rhatdan** [comment]
LGTM
@containers/podman-maintainers PTAL

**Review by vrothberg — COMMENTED** [review]
LGTM but I want a head nod from @mheon and/or @baude on the lock change

**Comment by mheon** [comment]
Moving the lock seems OK, shouldn't break anything.
/lgtm

**Pull request merged**